### DISCUSSION TITLE
Confusing use of of `#[\Closure(...)] attribute in the compiled container

### DISCUSSION THREAD

**Discussion by kaznovac** [symfony/symfony#49891] [Q&A]
This is the commit introducing it https://github.com/symfony/symfony/commit/d4e60e9d96a73883de937925cff61fb02204ae7f and the PR https://github.com/symfony/symfony/pull/45184 and its decision to remove the DI Closure attribute class https://github.com/symfony/symfony/pull/45184#discussion_r792666793

The FQCN points to the https://www.php.net/manual/en/class.closure.php, which is neither attribute nor its constructor has the stated parameters.

Here's its use: https://github.com/symfony/symfony/blob/d4e60e9d96a73883de937925cff61fb02204ae7f/src/Symfony/Component/EventDispatcher/Debug/WrappedListener.php#L126-L141

It just feels awkward to highjack the internal PHP class to carry the meta-information :|

**Comment by MatTheCat** [comment]
Well you linked the rationale behind this :grinning: 

> not having a class might be the best way to keep [this attribute] internal

Note that this is perfectly valid as long as `ReflectionAttribute::newInstance` is not called (see https://www.php.net/manual/en/language.attributes.reflection.php).

**Reply by kaznovac** [reply]
rationale does not hold: that is **WRONG** assumption that [`\Closure`](https://www.php.net/manual/en/class.closure.php) class does not exist

this code uses the existing PHP \Closure class (since 5.3) which is not marked to as attribute with wrong parameters to construct it 😕
